source/agora/node/Config.d

```diff
@@ -463,9 +463,20 @@ public struct RegistryConfig
                    "registry.{}: Authoritative zones require both 'primary' and " ~
                    "'email' fields to be set", name);
 
-            // If we're not primary, there's no configuration to validate
-            if (!zone.primary.set)
+            // Secondary registry
+            if (!zone.primary.set && zone.authoritative.set)
+            {
+                ensure(zone.primary_servers.length > 0,
+                    "registry.{} is secondary and at least one primary is required",
+                    name);
```

Comment:
Hold on... We have `primary`, `primary_servers`, and `redirect_primary` ? That's a bit much, isn't it ?

Reply:
Yes, when I was started `primary` wasn't there, thus I need to think over that again. `primary_servers` can have multiple servers configured as a backup.

Reply:
Currently, `primary` is for NS record which is cannot cover secondary servers right now (thinking a way)
`primary_servers` are list of DNS servers for a secondary to perform AXFR from; if one is not reachable second one will be tried
`redirect_primary` must be an TCP/HTTP endpoint that has `postValidator/postFlashNode` available for a secondary server. Thus this cannot replace `primary` directly; but I am thinking for `primary` and `primary_servers` right now.